plugin/evm/validators_cache_test.go

```diff
@@ -0,0 +1,84 @@
+package evm
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/flare-foundation/flare/ids"
+)
+
+type testData map[uint64]map[ids.ShortID]uint64
+
+func (testData) ByEpoch(uint64) (map[ids.ShortID]uint64, error) {
+	return map[ids.ShortID]uint64{}, nil
+}
+
+func TestWithCacheSize(t *testing.T) {
+	wantCacheSize := uint(42)
+
+	cfg := CacheConfig{}
+	WithCacheSize(wantCacheSize)(&cfg)
+
+	assert.Equal(t, wantCacheSize, cfg.CacheSize)
+}
+
+func TestValidatorsCache_ByEpoch(t *testing.T) {
+	var (
+		idsList []ids.ShortID
+		weights []uint64
+
+		cacheTestData = testData{}
+	)
+
+	numValidators := 6
+	for i := 0; i < numValidators; i++ {
+		id := randID()
+		weight := rand.Uint64()
+
+		idsList = append(idsList, id)
+		weights = append(weights, weight)
+		epoch := uint64(0)
+		if i > numValidators / 2 {
+			epoch = uint64(1)
+		}
+		cacheTestData[epoch] = map[ids.ShortID]uint64{
+			id: weight,
+		}
+	}
+
+	validatorsCache := NewValidatorsCache(cacheTestData, WithCacheSize(32))
+	for k, v := range cacheTestData {
+		validatorsCache.cache.Add(k, v)
+	}
+
+	t.Run("nominal case", func(t *testing.T) {
+		for i, epoch := range epochs {
```

Comment:
Where exactly is this `epochs` defined? I can't find it.

Reply:
In one of my suggestions I proposed that we remove the slice that was `epochs := []uint64{1, 2}` and I guess this is a remainder of it. I'll make a suggestion to improve this loop.